### PR TITLE
feat(storybook): add locale switcher - FE-4251

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour
Adds a locale switcher to the top toolbar in Storybook

Zrzut ekranu 2021-09-16 o 11 36 51
### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in sandbox/storybook
- [ ] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs

### Testing instructions

Run Storybook locally
or use [sandbox](https://codesandbox.io/s/carbon-quickstart-forked-0g6np?file=/src/index.tsx)

### CODE REVIEW

File: .storybook/locale-selector/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+
+import I18nProvider from "../../src/components/i18n-provider";
+import enGB from "../../src/locales/en-gb";
+import plPL from "../../src/locales/pl-pl";
+
+export const withLocaleSelector = (Story, context) => {
```
Comment (edleeks87) [inline .storybook/locale-selector/index.js:7]: Is there a specific reason you used a named export?

**Review by edleeks87 — DISMISSED** [review]

**Review by mmaciorowski-sage — APPROVED** [review]

**Review by mmaciorowski-sage — APPROVED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

File: src/locales/pl-pl.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
+export const isSingular = (count) => parseInt(count, 10) === 1;
```
Comment (sjohnson-sage) [inline src/locales/pl-pl.js:1]: I think this needs to be the same as the `isSingular` function in `en-gb.js`, only passing `count` into `parseInt` if it's a string

Comment (sjohnson-sage) [reply src/locales/pl-pl.js:1]: For more context `count` can be a string or a number, and `parseInt` can only accept a string as it's first parameter, so trying to use this in TypeScript is actually invalid:

image

**Review by sjohnson-sage — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by jamime — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**